packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -55,34 +70,80 @@ contract AnchorStateRegistry is Initializable, IAnchorStateRegistry, ISemver {
         return DISPUTE_GAME_FACTORY;
     }
 
-    /// @inheritdoc IAnchorStateRegistry
-    function tryUpdateAnchorState() external {
-        // Grab the game and game data.
-        IFaultDisputeGame game = IFaultDisputeGame(msg.sender);
-        (GameType gameType, Claim rootClaim, bytes memory extraData) = game.gameData();
+    /// @notice Verifies that the output root proposed as the root claim of a dispute game corresponds to the claimed
+    ///         L2 block number.
+    /// @param _disputeGame The dispute game contract.
+    /// @param _outputRootProof The output root proof corresponding to the proposed output root in the dispute game.
+    /// @param _headerRLP The RLP-encoded block header corresponding to the L2 block in the output root proof.
+    function verifyAnchor(
+        IFaultDisputeGame _disputeGame,
+        Types.OutputRootProof calldata _outputRootProof,
+        bytes calldata _headerRLP
+    )
+        external
+    {
+        (GameType gameType, Claim rootClaim, bytes memory extraData) = _disputeGame.gameData();
+        (uint256 l2BlockNumber) = abi.decode(extraData, (uint256));
 
         // Grab the verified address of the game based on the game data.
-        // slither-disable-next-line unused-return
         (IDisputeGame factoryRegisteredGame,) =
             DISPUTE_GAME_FACTORY.games({ _gameType: gameType, _rootClaim: rootClaim, _extraData: extraData });
 
         // Must be a valid game.
         require(
-            address(factoryRegisteredGame) == address(game),
+            address(factoryRegisteredGame) == address(_disputeGame),
             "AnchorStateRegistry: fault dispute game not registered with factory"
         );
 
         // No need to update anything if the anchor state is already newer.
-        if (game.l2BlockNumber() <= anchors[gameType].l2BlockNumber) {
-            return;
-        }
+        require(
+            l2BlockNumber > anchors[gameType].l2BlockNumber,
+            "AnchorStateRegistry: block number of proposal does not advance anchor"
+        );
 
         // Must be a game that resolved in favor of the state.
-        if (game.status() != GameStatus.DEFENDER_WINS) {
-            return;
+        require(
+            _disputeGame.status() == GameStatus.DEFENDER_WINS,
+            "AnchorStateRegistry: status of proposal is not DEFENDER_WINS"
+        );
+
+        require(
+            _disputeGame.resolvedAt().raw() + DISPUTE_GAME_FINALIZATION_DELAY.raw() <= block.timestamp,
```

Comment:
This contract now is looking like the OptimismPortal.
Why didn't we have this check before? Is that because we'd upgrade the ASR if it were to get poisoned by a blacklisted game? If so, why add the check now?